### PR TITLE
Ignore range tombstones during object table read

### PR DESCRIPTION
**Pull request by sadhansood**
When reading from objects table after invoking pruning, we run into rocksdb range tombstone fragmentation which ends up using significant cpu resources. RocksDB allows for an option (`ignore_range_deletion`) to disable the use of range tombstone iterator (which does fragmentation) during reads. So, we now setup `objects` table default read option to use `ignore_range_deletion` set as true going forward. The flip side is after pruning the object table of older object versions, those are still going to be visible to db read until compaction. But that is fine since we only ever read the latest object version (which is never pruned).

### CODE REVIEW

**Review by phoenix-o — APPROVED** [review]

File: crates/sui-core/src/authority/authority_store_pruner.rs [hunk 1]
```diff
@@ -326,4 +340,105 @@ mod tests {
         ma::assert_le!(after_compaction_size, before_compaction_size);
         Ok(())
     }
+
+    #[tokio::test]
+    async fn ensure_tombstone_fragmentation_in_stack_frame() -> Result<(), anyhow::Error> {
+        // This test writes a bunch of objects to objects table, invokes pruning on it and
+        // then does a bunch of get(). We open the db with `ignore_range_delete` set to false.
+        // We then record a cpu profile of the `get()` calls and find a range fragmentation stack frame
+        // in it.
```
Comment (longbowlu) [inline crates/sui-core/src/authority/authority_store_pruner.rs:349]: the two new test cases share a lot of duplicate code, can we refactor them?

File: crates/sui-storage/src/lib.rs [hunk 2]
```diff
@@ -19,7 +21,10 @@ pub fn default_db_options(
     cache_capacity: Option<usize>,
 ) -> (DBOptions, DBOptions) {
     let mut db_options = db_options
-        .map(|o| DBOptions { options: o })
+        .map(|o| DBOptions {
+            options: o,
+            rw_options: ReadWriteOptions::default(),
```
Comment (longbowlu) [inline crates/sui-storage/src/lib.rs:26]: why do we need this if it's the default value?

File: crates/sui-core/src/authority/authority_store_pruner.rs (hunk 1 above)
Comment (sadhansood) [reply crates/sui-core/src/authority/authority_store_pruner.rs]: sure, fixed it

File: crates/sui-storage/src/lib.rs (hunk 2 above)
Comment (sadhansood) [reply crates/sui-storage/src/lib.rs:26]: hmm, not sure what is the rusty way of doing it - i can look around and fix it later

**Pull request merged**